s3/s3.go

```diff
@@ -577,3 +438,23 @@ func hasCode(err error, code string) bool {
 	s3err, ok := err.(*Error)
 	return ok && s3err.Code == code
 }
+
+// requestRetryLoop attempts to send the request until the given
+// strategy says to stop.
+func requestRetryLoop(req *http.Request, retryStrat aws.AttemptStrategy) (*http.Response, error) {
+
+	for attempt := attempts.Start(); attempt.Next(); {
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			if shouldRetry(err) && attempt.HasNext() {
+				continue
+			}
+			return nil, err
+		}
+
+		return resp, nil
+	}
+
+	panic("could not complete the request within the specified retry attempts")
```

Comment:
This was the pattern established when I began working on the code, but you're correct. I've switched it to an error.